cmake/FindArpack.cmake

```diff
@@ -0,0 +1,24 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+if(NOT ARPACK_ROOT)
+  # Need to set to empty to avoid warnings with --warn-uninitialized
+  set(ARPACK_ROOT "")
+  set(ARPACK_ROOT $ENV{ARPACK_ROOT})
+endif()
+
+find_path(ARPACK_INCLUDE_DIR arpack/arpackdef.h
```

Comment:
I think checking for `arpack/arpack.hpp` here is the right thing to do, which is the file you would include for the C++ bindings, no?

Reply:
Done.
We are not using the c++ bindings provided by arpack-ng because there are quite recent bug fixes in the git repo. We are calling Arpack directly `src/NumericalAlgorithms/LinearAlgebra/FindGeneralizedEigenvaluesArpack.cpp` #3431.

Reply:
If you're not using the C++ bindings, why check for them?

Reply:
I think it's better to find the include dir, then we have the possibility to use the includes. Which header we look for is probably not super important. My suggestion is to leave as-is.

I don't see any recent commits, issues or PRs in the arpack-ng repo mentioning bugs, which are you referring to?

Reply:
I reverse my position. Reading https://github.com/opencollab/arpack-ng/blob/master/CMakeLists.txt it looks like C++ bindings are not built by default. I agree then, that it's probably easiest to not rely on them and you can delete the include-dir find in cmake after all. I find this quite unfortunate though :(

Reply:
My distro packages arpack-ng, but doesn't install the C++ bindings.  If we're not using the C++ bindings, I'd rather not be forced to re-package the thing just to satisfy a CMake check.

Reply:
(Collision on the last two comments.)  I will say that I don't object to testing for the C bindings, since they should presumably be everywhere.  I also don't object to not testing for any headers, since we don't use them.

Reply:
Agreed, see above. Thanks for checking this @wthrowe.

Reply:
It looks like by default arpack installs no useful headers, so let's not try to find any. Even `spack install arpack-ng` doesn't install any.

Reply:
I will go ahead and remove the include-dir part from the cmake file.


@nilsleiffischer I was referring to bug fix commits in the ICB part of the code https://github.com/opencollab/arpack-ng/tree/master/ICB. There are quite a few in the last two years. The official ubuntu20.04 package for arpack(https://ubuntu.pkgs.org/20.04/ubuntu-universe-arm64/libarpack2-dev_3.7.0-3_arm64.deb.html) is 3years old which means a lot of these fixes may not be present.
Note that these binding are something that arpack-ng people have added in the last few years, the Arpack FORTRAN code that we calling directly is quite old and well tested.

Reply:
Thanks for explaining 👍